src/template_utils.rs

```diff
@@ -9,13 +15,36 @@ pub static CACHE_NAME: &str = env!("CACHE_ID");
 
 pub type BaseContext<'a> = &'a(&'a Connection, &'a Catalog, Option<User>);
 
-pub type Ructe = Content<Vec<u8>>;
+#[derive(Debug)]
+pub struct Ructe(pub Vec<u8>);
+
+impl<'r> Responder<'r> for Ructe {
+    fn respond_to(self, r: &Request) -> response::Result<'r> {
+        //if method is not Get or page contain a form, no caching
+        if r.method() != Method::Get || self.0.windows(6).any(|w| w == b"<form ") {
```

Comment:
Probably obvious, but why do we disable caching on pages with a form?

Reply:
If the page contain a form, the csrf token will be cached, and may no longer match user's cookies. As these tokens are added later by a fairing, there is no way to detect the page changed here. The other solution could be to allow caching of page and for rocket_csrf to remove ETag when appropriate